streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -78,26 +78,36 @@
 
     private final Serde<K> keySerde;
     private final Serde<V> valSerde;
+    private final String storeName;
 
     private boolean sendOldValues = false;
 
+
     public KTableImpl(KStreamBuilder topology,
                       String name,
                       ProcessorSupplier<?, ?> processorSupplier,
-                      Set<String> sourceNodes) {
-        this(topology, name, processorSupplier, sourceNodes, null, null);
+                      Set<String> sourceNodes,
+                      final String storeName) {
+        this(topology, name, processorSupplier, sourceNodes, null, null, storeName);
     }
 
     public KTableImpl(KStreamBuilder topology,
                       String name,
                       ProcessorSupplier<?, ?> processorSupplier,
                       Set<String> sourceNodes,
                       Serde<K> keySerde,
-                      Serde<V> valSerde) {
+                      Serde<V> valSerde,
+                      final String storeName) {
         super(topology, name, sourceNodes);
         this.processorSupplier = processorSupplier;
         this.keySerde = keySerde;
         this.valSerde = valSerde;
+        this.storeName = storeName;
```

Comment:
Yeah we should for store name, and we can use `Utils.notNull`.

For others like serdes, since they are nullable we should not check.